### PR TITLE
Correct the mapping on external resources links in supplementary information page

### LINKED ISSUE #195: Correct the mapping on external resources links in Supplementary Information page
We have some conflicts mapping issues in bulk Supplementary Information page, regarding experiment type and ArrayExpress. 

In the case of `E-PROT-39`, as its experiment type is `RNASEQ_MRNA_DIFFERENTIAL` so the external resources are grouped to `ENA` and it also contains `ArrayExpress` link which is invalid either.

https://www.ebi.ac.uk/gxa/experiments/E-PROT-39/Supplementary%20Information

### PR DESCRIPTION
**Pull request by lingyun1010**
This is a big change on external resources links, basically we would like to decouple the logic from experiment type and the external resources. The detailed rules are given by Silvie stated in the related issue.

https://github.com/ebi-gene-expression-group/atlas-web-bulk/issues/195

### CODE REVIEW

File: app/src/main/java/uk/ac/ebi/atlas/experimentpage/ExpressionAtlasContentService.java [hunk 1]
```diff
@@ -174,73 +131,83 @@ public ImmutableList<ExternallyAvailableContent> list(String experimentAccession
                                                  String accessKey,
                                                  ExternallyAvailableContent.ContentType contentType) {
         Experiment<?> experiment = experimentTrader.getExperiment(experimentAccession, accessKey);
-        String externalResourceType = externalResourceLinksPriority(experiment);
         ImmutableList.Builder<ExternallyAvailableContent> arrayExpressAndOtherExternalResourcesLinks = ImmutableList.builder();
```
Comment (ke4) [inline app/src/main/java/uk/ac/ebi/atlas/experimentpage/ExpressionAtlasContentService.java]: We can just name this as `externalResourcesLinks`.

```suggestion
        var externalResourcesLinks = ImmutableList.builder();
```

**Review by ke4 — CHANGES_REQUESTED** [review]
Just a tiny one. Thanks for your work

**Review by ke4 — CHANGES_REQUESTED** [review]
As I mentioned already in your other related PRs: you need to fix `atlas-web-core` submodule to base on the latest changes.
Thanks

**Comment by ke4** [comment]
This PR also contains some of the commits twice.
Could you clear this, please?
Thanks

**Review by ke4 — COMMENTED** [review]
This PR also contains some of the commits twice.
Could you clear this, please?
Thanks

**Comment by lingyun1010** [comment]
> This PR also contains some of the commits twice. Could you clear this, please? Thanks

My git is doing something duplicate ones automatically when I rebase on the atlas-develop-core `develop` branch, otherwise, even though I updated the atlas-web-core, the github checks still got a conflict issue ( like I mentioned here: https://github.com/ebi-gene-expression-group/atlas-web-single-cell/pull/427#issuecomment-2220560900). So I think, I created a new rebased branch on the latest atlas-web-core submodule branch, without my previous commits and then merged into this one.

**Review by ke4 — DISMISSED** [review]
LGTM

**Review by ke4 — DISMISSED** [review]
LGTM 2nd try

**Review by ke4 — DISMISSED** [review]
3rd try: LGTM!

**Pull request merged**